### PR TITLE
feat: implement artifact db cache for cost surfaces

### PR DESCRIPTION
**Pull request by Dyostiq**
### Overview

_Please write a description. If the PR is hard to understand, provide a quick explanation of the code._

### Designs

_Link to the related design prototypes (if applicable)_

### Testing instructions

_Please explain how to test the PR: ID of a dataset, steps to reach the feature, etc._

### Feature relevant tickets

_Link to the related task manager tickets_

---

## Checklist before submitting

- [ ] Meaningful commits and code rebased on `develop`.
- [ ] If this PR adds feature that should be tested for regressions when
      deploying to staging/production, please add brief testing instructions
      to the deploy checklist (`docs/deployment-checklist.md`)
- [ ] Update CHANGELOG file

### CODE REVIEW

File: api/libs/scenario-cost-surface/src/cost-surface-file-cache.api.entity.ts
Comment (kgajowy) [inline api/libs/scenario-cost-surface/src/cost-surface-file-cache.api.entity.ts:49]: is it an entity responsibility to decide which files are under cache? 🤔

Comment (Dyostiq) [reply api/libs/scenario-cost-surface/src/cost-surface-file-cache.api.entity.ts:49]: the const is defining only which fields are needed to create an instance

File: api/libs/scenario-cost-surface/src/cost-surface-file-cache.api.entity.ts
Comment (Dyostiq) [reply api/libs/scenario-cost-surface/src/cost-surface-file-cache.api.entity.ts:24]: AFAIR the generator does not work, so yes, it is written by hand.
The scenario id is a key here, artifacts are generated per scenario. I can drop the reference, but then the validation and cascading are dropped. Honestly, I don't have a strong opinion on that.

Moving the scenario is rather not feasible, as it has other references and the entire graph would land in the lib.

Comment (kgajowy) [reply api/libs/scenario-cost-surface/src/cost-surface-file-cache.api.entity.ts:24]: May want to just drop a note then why the difference is there. I am fine with the pros/cons.

File: api/libs/scenario-cost-surface/src/cost-surface-file-cache.api.entity.ts
Comment (kgajowy) [reply api/libs/scenario-cost-surface/src/cost-surface-file-cache.api.entity.ts:49]: just for a sake of security of typing, shouldn't they relate to actual fields?

File: api/libs/scenario-cost-surface/src/cost-surface-file-cache.api.entity.ts
Comment (hotzevzl) [reply api/libs/scenario-cost-surface/src/cost-surface-file-cache.api.entity.ts:24]: we've not been using entities as source of truth so far, so it's ok to have the migration be the source of truth here

File: api/libs/scenario-cost-surface/src/cost-surface-file-cache.api.entity.ts
Comment (hotzevzl) [reply api/libs/scenario-cost-surface/src/cost-surface-file-cache.api.entity.ts:49]: it should be ok as https://github.com/Vizzuality/marxan-cloud/pull/228/files/14076746b22f7c4e320e8e6473503343e85b8ba3#diff-0298940b2ab35a3d32a4536133f253445f3c2ce55b768011266c254e8fd80596R42 would bark if we used this array including in it elements which don't match props of `CostSurfaceFileCache`

Comment (Dyostiq) [reply api/libs/scenario-cost-surface/src/cost-surface-file-cache.api.entity.ts:49]: I've added a `type check` for that

**Review by kgajowy — APPROVED** [review]

**Review by hotzevzl — APPROVED** [review]

**Comment by hotzevzl** [comment]
@Dyostiq feel free to merge this after merging its base branch if you are happy with it and don't need to apply any further changes

**Pull request closed without merging**
